### PR TITLE
fix: RPC events sync.

### PR DESCRIPTION
**Pull request by momintlh**
- RPC events name sync between players. #68 
- Made Players dictionary private #66

Temporary fix, RPC part needs to be refactored, might be better to handle within JS. #40

### CODE REVIEW

File: Assets/PlayroomKit/PlayroomKit.cs [hunk 1]
```diff
@@ -1078,6 +1078,15 @@ private static void InvokeRpcRegisterCallBack(string dataJson, string senderJson
                 Debug.LogError(ex.Message);
             }
 
+
```
Comment (SaadBazaz) [inline Assets/PlayroomKit/PlayroomKit.cs:1081]: A bit of description Why this needs to be done would be great, somewhere in the code. So the future you can send blessings on the current you.

**Review by SaadBazaz — CHANGES_REQUESTED** [review]
Either my eyes are deceiving me, or I can't see the `private Players` change.

File: Assets/PlayroomKit/PlayroomKit.cs (hunk 1 above)
Comment (SaadBazaz) [reply Assets/PlayroomKit/PlayroomKit.cs:1081]: @momintlh Could you add a description? This implementation could easily be forgotten in the future.

**Review by SaadBazaz — CHANGES_REQUESTED** [review]
- Add description comments
- Write dry runs on all cases in mind (no need to actually build and run)

File: Assets/PlayroomKit/PlayroomKit.cs [hunk 2]
```diff
@@ -1095,6 +1105,9 @@ private static void InvokeRpcRegisterCallBack(string dataJson, string senderJson
 
         public static void RpcCall(string name, object data, RpcMode mode, Action callbackOnResponse)
         {
+            // setting to sync the event name with other players
```
Comment (SaadBazaz) [inline Assets/PlayroomKit/PlayroomKit.cs:1108]: This is called a "parrot comment"; the comment is describing the same thing the code describes. Usually, a comment should be more of the "Why?" than the "What?". The What can be understood just by reading the code.

**Review by SaadBazaz — APPROVED** [review]

**Comment by SaadBazaz** [comment]
I have a hunch this methodology might cause Race conditions upon high load. Needs testing on high performance calls between **multiple players,** with multiple RPCs.

Instead of saving a state, maybe send some metadata **alongside each RPC call?** That way, it's guaranteed to reach along with the RPC call.

File: Assets/PlayroomKit/PlayroomKit.cs [hunk 3]
```diff
@@ -1111,6 +1121,19 @@ public static void RpcCall(string name, object data, RpcMode mode, Action callba
                 }
             }
 
+
+            JSONArray jsonArray = new JSONArray();
+            foreach (string item in rpcCalledEvents)
+            {
+                jsonArray.Add(item);
```
Comment (SaadBazaz) [inline Assets/PlayroomKit/PlayroomKit.cs:1128]: Can this overflow, as you're not checking for duplicates or dead references?

**Review by SaadBazaz — CHANGES_REQUESTED** [review]
What scenarios has this been tested on?

File: Assets/PlayroomKit/PlayroomKit.cs (hunk 3 above)
Comment (momintlh) [reply Assets/PlayroomKit/PlayroomKit.cs:1128]:
```c#
  if (!rpcCalledEvents.Contains(name))
  {
      rpcCalledEvents.Add(name);
  }
```

Duplicates aren't added to the list, this check if before the JSON serialization.

**Comment by SaadBazaz** [comment]
Merging in favor of a working solution instead of over engineering. Noted down problems in Refactor Umbrella issue.

**Pull request merged**